File: src/main/java/io/github/emcw/adapters/ColorAdapter.java
package io.github.emcw.adapters;

import com.google.gson.*;

import java.awt.*;
import java.lang.reflect.Type;

public class ColorAdapter implements JsonSerializer<Color>, JsonDeserializer<Color> {
    @Override
    public JsonElement serialize(Color src, Type srcType, JsonSerializationContext ctx) {
        String colorString = "#" + Integer.toHexString(src.getRGB()).substring(2);
        return new JsonPrimitive(colorString);
    }

    @Override
    public Color deserialize(JsonElement json, Type type, JsonDeserializationContext ctx) throws JsonParseException {
        return Color.decode(json.getAsString());
    }
}

File: src/main/kotlin/io/github/emcw/adapters/ColorAdapter.kt
package io.github.emcw.adapters

import com.google.gson.*
import java.awt.Color
import java.lang.reflect.Type

class ColorAdapter : JsonSerializer<Color>, JsonDeserializer<Color> {
    override fun serialize(src: Color, srcType: Type, ctx: JsonSerializationContext): JsonElement {
        val colorString = "#" + Integer.toHexString(src.rgb).substring(2)
        return JsonPrimitive(colorString)
    }

    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, type: Type, ctx: JsonDeserializationContext): Color {
        return Color.decode(json.asString)
    }
}